cmake/kokkos_compiler_id.cmake

```diff
@@ -153,7 +157,7 @@ ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA)
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
   SET(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "Kokkos turns off CXX extensions" FORCE)
-ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL HIP)
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
```

Comment:
Any reason for the change from `HIP` to `HIPCC`?

Reply:
Because effectively this tells whether hipcc is being used as the compiler.  "hip" does not designate any compiler.